lib/rules/number-no-partial-percent/index.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const balancedMatch = require('balanced-match');
+
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+
+const ruleName = 'number-no-partial-percent';
+
+const messages = ruleMessages(ruleName, {
+	invalidPercent: 'Percent values must be 0% or 100%',
+});
+
+const meta = {
+	url: 'https://stylelint.io/user-guide/rules/no-partial-percent',
+};
+
+/** @type {import('stylelint').Rule} */
+const rule = (primary) => {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual: primary });
+
+		if (!validOptions) {
+			return;
+		}
+
+		root.walkDecls((decl) => {
+			const value = decl.value;
+
+			if (!value.includes('%')) return;
+
+			// Match that start of color functions and percent values
+			const matches = value.matchAll(/(?:rgba?|color)\s*\(|[\d.]+%/g);
```

Comment:
question

Is the list of functions exhaustive?

Reply:
I went over all the CSS functions that exist and made sure this list is exhaustive

Reply:
I went over all the CSS functions that exist and made sure this list is exhaustive